cstar_ocean/case.py

```diff
@@ -18,6 +16,9 @@
     SurfaceForcing,
 )
 
+if TYPE_CHECKING:
+    from cstar_ocean.base import Component, BaseModel
```

Comment:
There are a lot of imports that are actually only needed for type hints. Putting these behind the `if TYPE_CHECKING` statement avoids circular imports, and makes it clearer when you actually need to import the class constructor and when you are only dealing with pre-instantiated instance of that class.

Reply:
(... but if you do this you need to use `"<ClassName>"`, not `<ClassName>`, which I forgot)